src/main/java/org/primefaces/component/datagrid/DataGridRenderer.java

```diff
@@ -170,20 +171,24 @@ protected void encodeGrid(FacesContext context, DataGrid grid) throws IOExceptio
         int itemsToRender = rows != 0 ? rows : grid.getRowCount();
         int numberOfRowsToRender = (itemsToRender + columns - 1) / columns;
         int displayedItemsToRender = rowIndex + itemsToRender;
-        String columnClass = DataGrid.COLUMN_CLASS + " " + GridLayoutUtils.getColumnClass(columns);
+        String columnClass = DataGrid.COLUMN_CLASS + " " + GridLayoutUtils.getColumnClass(columns) + " " + grid.getRowStyleClass();
+        String columnInlineStyle = grid.getRowStyle();
+
+        writer.startElement("div", null);
+        writer.writeAttribute("class", DataGrid.GRID_ROW_CLASS, null);
```

Comment:
By moving this outside the loop doesn't that mean only 1 row will ever be created?  this looks like it was meant to create a DIV per row?

Reply:
This IMO was necessary for a very old grid system before PrimeFaces 6.0 introduced Grid CSS.
IMO DataGrid was adapted to Grid CSS at some point, but this was never cleaned up.

Reply:
... and i had to modify this because with ui-g over each row there was no way to add additional responsiveness.
Check this against the showcase. :-)

Reply:
perfect